File: app/src/main/java/vn/vistark/giam_sat_nha_yen/ui/dashboard_screen/timer_list/TimerListViewHolder.java
package vn.vistark.giam_sat_nha_yen.ui.dashboard_screen.timer_list;

import android.view.View;
import android.widget.RelativeLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.appcompat.widget.SwitchCompat;
import androidx.recyclerview.widget.RecyclerView;

import vn.vistark.giam_sat_nha_yen.R;
import vn.vistark.giam_sat_nha_yen.ui.dashboard_screen.DashboardScreenActivity;

class TimerListViewHolder extends RecyclerView.ViewHolder {
    private TextView tvTimerItemId;
    private TextView tvTimerItemLabel;
    private TextView tvTimerItemStart;
    private TextView tvTimerItemEnd;
    SwitchCompat scTimerSwitchCompat;
    RelativeLayout rlTimerItemRoot;

    TimerListViewHolder(@NonNull View v) {
        super(v);
        initViews(v);
    }

    private void initViews(View v) {
        tvTimerItemId = v.findViewById(R.id.timerItemId);
        tvTimerItemLabel = v.findViewById(R.id.timerItemLabel);
        tvTimerItemStart = v.findViewById(R.id.timerItemStart);
        tvTimerItemEnd = v.findViewById(R.id.timerItemEnd);
        scTimerSwitchCompat = v.findViewById(R.id.timerSwitchCompat);
        rlTimerItemRoot = v.findViewById(R.id.timerItemRoot);
    }

    void updateTimerItemIdView(final String id) {
//        tvTimerItemId.post(new Runnable() {
//            @Override
//            public void run() {
//            }
//        });
        tvTimerItemId.setText(id);
    }

    void updateTimerItemLabelView(final String label) {
//        tvTimerItemLabel.post(new Runnable() {
//            @Override
//            public void run() {
//
//            }
//        });
        tvTimerItemLabel.setText(label);
    }

    void updateTimerItemStartView(final String start) {
//        tvTimerItemStart.post(new Runnable() {
//            @Override
//            public void run() {
//
//            }
//        });
        tvTimerItemStart.setText(start);
    }

    void updateTimerItemEndView(final String end) {
//        tvTimerItemEnd.post(new Runnable() {
//            @Override
//            public void run() {
//
//            }
//        });
        tvTimerItemEnd.setText(end);
    }

    void updateTimerSwitchCompatView(final boolean state) {
//        scTimerSwitchCompat.post(new Runnable() {
//            @Override
//            public void run() {
//
//            }
//        });
        scTimerSwitchCompat.setChecked(state);
    }

    void toggleTimerSwitchCompatView() {
//        scTimerSwitchCompat.post(new Runnable() {
//            @Override
//            public void run() {
//
//            }
//        });
        scTimerSwitchCompat.toggle();
    }

    void updateTimerStateView(final boolean state) {
//        tvTimerItemId.post(new Runnable() {
//            @Override
//            public void run() {
//
//            }
//        });
        if (state) {
            tvTimerItemId.setBackgroundResource(R.drawable.quepal);
        } else {
            tvTimerItemId.setBackgroundResource(R.drawable.burning_orange);
        }
    }

    void holderUpdateEmptyNotify(int size) {
        if (size > 0) {
            ((DashboardScreenActivity) tvTimerItemId.getContext()).hideTimerListEmptyNotify();
        } else {
            ((DashboardScreenActivity) tvTimerItemId.getContext()).showTimerListEmptyNotify();
        }
    }
}

File: app/src/main/java/vn/vistark/dknncy/ui/dashboard_screen/timer_list/TimerListViewHolder.kt
package vn.vistark.dknncy.ui.dashboard_screen.timer_list

import android.view.View
import android.widget.RelativeLayout
import android.widget.TextView
import androidx.appcompat.widget.SwitchCompat
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import vn.vistark.dknncy.R
import vn.vistark.dknncy.data.firebase.FirebaseConfig
import vn.vistark.dknncy.data.modal.TimerItem
import vn.vistark.dknncy.ui.dashboard_screen.DashboardScreenActivity

class TimerListViewHolder(v: View) : ViewHolder(v) {
    private var tvTimerItemId: TextView = v.findViewById(R.id.timerItemId)
    private var tvTimerItemLabel: TextView = v.findViewById(R.id.timerItemLabel)
    private var tvTimerItemStart: TextView = v.findViewById(R.id.timerItemStart)
    private var tvTimerItemEnd: TextView = v.findViewById(R.id.timerItemEnd)
    var scTimerSwitchCompat: SwitchCompat = v.findViewById(R.id.timerSwitchCompat)
    var rlTimerItemRoot: RelativeLayout = v.findViewById(R.id.timerItemRoot)

    var context = tvTimerItemId.context as DashboardScreenActivity

    fun bind(ti: TimerItem) {
        tvTimerItemId.text = ti.id.toString()
        tvTimerItemId.isSelected = true
        tvTimerItemLabel.text = ti.label
        tvTimerItemLabel.isSelected = true
        tvTimerItemStart.text = ti.start
        tvTimerItemEnd.text = ti.end
        scTimerSwitchCompat.isChecked = ti.isPower
        tvTimerItemId.setBackgroundResource(if (ti.isPower && ti.isState) R.drawable.quepal else R.drawable.burning_orange)

        onSweetChange(ti)
    }

    private fun onSweetChange(ti: TimerItem) {
        // Sự kiện khi đổi trạng thái bật nguồn hoặc tắt nguồn thê thục hiện hoặc bỏ qua timer
        scTimerSwitchCompat.setOnClickListener {
            ti.isPower = !ti.isPower
            println(">>>>>>> Đã gạt thành [${ti.isPower}] cho ID: ${ti.id}")
            FirebaseConfig.updateData(ti)
        }
//        scTimerSwitchCompat.setOnCheckedChangeListener { compoundButton, b ->
//            ti.isPower = b
//            println(">>>>>>> Đã gạt thành [$b] cho ID: ${ti.id}")
//            FirebaseConfig.updateData(ti)
//        }
    }
}